cadence: fix wrong wrappings

Melkor333 opened the pull request:
###### Motivation for this change
This fixes #48790

Some moduls of the Packages were automatically wrapped, but the problem is that not all of them need to be wrapped and there are some name mismatches. E.g. one Package is called `cadence-logs` in the binaries but `logs.py` in the shared folder (instead of `cadence-logs.py`). also `-` in the bin folder is `_` in the shared folder.

Just to point out, there needs more to be done to make this package really usable, I just don't know how to do the fixes:

- almost all binaries immediately stop with the error message that jack is either not installed or can't be connected to (no matter if jack runs or not). I think this is because they all need `libjack.so.0`. which somehow is not available (The file `share/cadence/src/jacklib.py` wants to load it).
 I tried to fix it by adding `libjack2` to the buildinputs, but it made no difference. If somebody can point out how I can add `libjack.so.0` to the path, I can add a fix in this pull request.
- most included tools try to import `dbus.mainloop.pyqt5` which seems to not exist in the `dbus-python` version available on nixos. E.g. the program `claudia`  stops immediately saying `DBus is not available`. I guess we need to build a package containing `dbus.mainloop.pyqt5` before we can fix this issue...

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Melkor333 committed:
cadence: fix wrong wrappings

c0bw3b committed:
cadence: split native build inputs

+ fix license : this is under GPL2+ and not MIT

c0bw3b commented:
Verified locally and everything seems in order between shell scripts, Python scripts and compiled binaries.

c0bw3b committed:
cadence: use fetchzip

because GitHub archives are not stable over time so better
to compute the hash of the unpacked content

c0bw3b commented:
@GrahamcOfBorg build cadence

GrahamcOfBorg commented:
<!--REQUEST_ID=97576a74-029d-4970-9048-058da476aef4-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50610&attempt_id=fbe0d813-8647-4f8e-99c1-5370df2d97e7)

Attempted: cadence

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence/share/cadence/src/catia.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/y78kh4c49qz3m8al5mmxqxanncfs4x0h-python3-3.7.1/bin/python3"
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence/share/cadence/src/shared.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/y78kh4c49qz3m8al5mmxqxanncfs4x0h-python3-3.7.1/bin/python3"
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence/share/cadence/src/claudia_launcher.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/y78kh4c49qz3m8al5mmxqxanncfs4x0h-python3-3.7.1/bin/python3"
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence/share/cadence/src/patchcanvas.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/y78kh4c49qz3m8al5mmxqxanncfs4x0h-python3-3.7.1/bin/python3"
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence/bin/cadence-pulse2loopback: interpreter directive changed from "/bin/bash" to "/nix/store/n2hjbpkf4c0m48945ivxs3lmsczzw2rg-bash-4.4-p23/bin/bash"
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence/bin/cadence-pulse2jack: interpreter directive changed from "/bin/bash" to "/nix/store/n2hjbpkf4c0m48945ivxs3lmsczzw2rg-bash-4.4-p23/bin/bash"
checking for references to /build in /nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence...
postPatchMkspecs
postPatchMkspecs
/nix/store/00abkzag1iqr6fiwbad4x2qm7il3gxb1-cadence
```
</p></details>

c0bw3b committed:
cadence: limit to x64-linux

Broken on aarch64 because the Makefile pass
"-msse -mfpmath=sse" flags (x86 only)

GrahamcOfBorg commented:
<!--REQUEST_ID=97576a74-029d-4970-9048-058da476aef4-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50610&attempt_id=9104dcf7-441a-433d-b9e9-fded2edb26db)

Attempted: cadence

<details><summary>Partial log (click to expand)</summary><p>

```
make[3]: Leaving directory '/build/qtwebengine-everywhere-src-5.11.1/src/core'
make[2]: *** [Makefile:80: sub-gn_run-pro-make_first] Error 2
make[2]: Leaving directory '/build/qtwebengine-everywhere-src-5.11.1/src/core'
make[1]: *** [Makefile:79: sub-core-make_first] Error 2
make[1]: Leaving directory '/build/qtwebengine-everywhere-src-5.11.1/src'
make: *** [Makefile:47: sub-src-make_first] Error 2
builder for '/nix/store/280x803mp96p5yyrg5lv83644g62v2ya-qtwebengine-5.11.1.drv' failed with exit code 2
cannot build derivation '/nix/store/2cypxvzsv9i7n6j3zsm9varz0sdcv8dl-python3.7-PyQt-5.10.1.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/p0w40c750hg5nkbwk8wbal7jxc1q9nf8-cadence.drv': 1 dependencies couldn't be built
error: build of '/nix/store/p0w40c750hg5nkbwk8wbal7jxc1q9nf8-cadence.drv' failed
```
</p></details>

c0bw3b commented:
Thanks @Melkor333
